txpool/txpool.go

```diff
@@ -672,6 +745,9 @@ func (t *txHeapWrapper) Promote() []*types.Transaction {
 	// Grab its nonce value and set it as the reference next nonce
 	t.nextNonce = lastTxn.Nonce + 1
 
+	// Save this tx in case of a imminent resubmit (speedUp)
+	t.lastPromoted = lastTxn
```

Comment:
Please give a better explanation as to why we need the `lastPromoted` field

Reply:
Basically, I wanted to avoid doing a search&find when removing the old transaction and have a constant lookup. (Because the moment a transaction is promoted to the `pendingQueue` it is no longer "tracked", so to speak)

[Documented](https://github.com/0xPolygon/polygon-sdk/pull/220/commits/8e4ce12c2c344c912148fb5a04a2c386774a0836)